pkg/kv/kvserver/logstore/logstore.go

```diff
@@ -543,6 +545,16 @@ func LoadEntries(
 
 	ents, cachedSize, hitIndex, exceededMaxBytes := eCache.Scan(ents, rangeID, lo, hi, maxBytes)
 
+	budgetFor := func(size uint64) bool {
+		return account == nil || account.Grow(ctx, int64(size)) == nil
+	}
+	// If we can't get quota for the cached entries, do not return them, even
+	// though they are already in memory. Returning them now increases their
+	// lifetime, and risks getting the server out of memory.
+	if !budgetFor(cachedSize) {
```

Comment:
Shouldn't we return the entries we do have budget for?

Reply:
Yeah, perhaps we can loop over the cached entries to grow the budget one by one.